Docs suggestion - Update database-usage.mdx for clarity on free tier and paid tier behaviour and read only for both free and paid (#12480)
supabase/supabase: apps/docs/pages/guides/platform/database-usage.mdx

## Database storage management

Supabase uses network-attached storage to balance performance with scalability.

> For Pro and Enterprise tier projects, [disk size expands ~1.5x automatically](#paid-tier-automatic-disk-resizing) when you reach 90% of the disk size. Disk size expansion can only occur once every six hours.

Reviewer: ```suggestion
For Pro and Enterprise tier projects, [disk size expands ~1.5x automatically](#paid-tier-disk-auto-scaling) when you reach 90% of the disk size. Disk size expansion can only occur once every six hours.
```

> For Pro and Enterprise tier projects, [disk size expands ~1.5x automatically](#paid-tier-automatic-disk-resizing) when you reach 90% of the disk size. Disk size expansion can only occur once every six hours.

Reviewer 2: I think there is duplicated content here. We talk about the 1.5x, 90%, 6 hours all below.

Author: Yea your right, i've changed it to bullet points to get the main info across with anchor links. Need a rereview if that's ok 😄

Author: fixed

For Free tier projects, the database will enter ['read only' mode when you reach 95% disk space](#free-tier-project-read-only-mode).

Pro and enterprise tier projects can store up to 1024TB.

### Free tier project 'read-only' mode

Free tier projects enter read-only mode when you reach 95% of the disk size.
In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
Regular operation (read-write mode) is automatically re-enabled once usage is below 95% of the disk size.

> For [automatic disk resizing](paid-tier-automatic-disk-resizing), and avoid your database entering read-only mode, you can [upgrade to the Pro or Enterprise plan](https://app.supabase.com/project/_/settings/billing/subscription).

Reviewer: ```suggestion
For [disk auto-scaling](paid-tier-disk-auto-scaling), and avoid your database entering read-only mode, you can [upgrade to the Pro or Enterprise plan](https://app.supabase.com/project/_/settings/billing/subscription).
```

> For [automatic disk resizing](paid-tier-automatic-disk-resizing), and avoid your database entering read-only mode, you can [upgrade to the Pro or Enterprise plan](https://app.supabase.com/project/_/settings/billing/subscription).

Reviewer 2: Again, I think this is covered in the section below (Increasing available disk size), needn't duplicate. And the below section suggests both alternatives - upgrading to pro or reduce disk usage.

Author: Think this is fixed now.

There are 2 distinct scenarios about this though, there is the situation in which someone has a free tier project ending up in read only, and also paid tier. You might need to re-review though.

> ### Paid tier automatic disk resizing

Reviewer: ```suggestion
### Paid tier disk auto-scaling
```

> If you are on Pro or Enterprise, database storage expands automatically when you reach 90% of the disk size. Usually the disk is expanded to be 50% larger (e.g., 8GB -> 12GB).

Reviewer: ```suggestion
Pro and Enterprise have disk auto-scaling. Storage expands automatically when the database reaches 90% of the disk size. The disk is expanded to be 50% larger (e.g., 8GB -> 12GB).
```

> For every automatic database storage expansion, it will not auto expand for another 6 hours.

Reviewer: ```suggestion
Auto-scaling can only take place once every 6 hours.
```

> If you intend to import a lot of data into your database that would require multiple storage expansions then [reach out to our team](https://app.supabase.com/support/new), otherwise your project might enter a ['read-only' mode](#paid-tier-project-read-only-mode).

Reviewer: I think this is already said below?

```
If you need more than 1024TB of disk size or require multiple storage expansions in a short period of time, [contact us](https://app.supabase.com/support/new) to learn more about the Enterprise plan.
```

Author: fixed

> If you need more than 1024TB of disk size or require multiple storage expansions in a short period of time, [contact us](https://app.supabase.com/support/new) to learn more about the Enterprise plan.

Author: This isn't a great suggestion, but we are talking about 2 different things here, so split it up.

```suggestion
You may require multiple storage expansions in a short period of time. For example, you need to upload more than 1.5x the current size of your database storage. This will lead to your database entering a [read-only mode](#paid-tier-project-read-only-mode). You can avoid this by [contacting the support team](https://app.supabase.com/support/new) so that your database can be resized to a sufficient amount before you start uploading to it.

If you need more than 1024TB of disk size, you can [contact us](https://app.supabase.com/support/new) to learn more about the Enterprise plan. 
```

I think we now also support self serve auto-scaling, or it's coming soon. so need to also update this.

Not that sure we need to keep talking about the 1024TB limit, it always sounds a bit like a negative enforced limit.

#### Paid tier project 'read-only' mode

> During the 6 hour period after an automatic resize, your database will not automatically resize until after 6 hours.

Reviewer: ```suggestion
Database storage expands, at most, once every 6 hours.
```

If within those 6 hours you reach 95% of the disk space, your project will enter read-only mode.

In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
Regular operation (read-write mode) is automatically re-enabled once usage is below 95% of the disk size, or when the database automatically resizes after the 6 hour period.

### Increasing available disk size

> 1. [Upgrade to the Pro or Enterprise plan](https://app.supabase.com/project/_/settings/billing/subscription) to increase your quota and expand your disk size automatically.

Reviewer 2: Can we clarify that this is an either or? either upgrade to pro to not worry about this or reduce your disk usage

Author: great point.

Author: i'll change this in a commit.

Author: need a rereview 😄
